### PR TITLE
New function auto_detect _indicators in common server

### PR DESCRIPTION
**Pull request by orhovy**
#6680  Status
- [x] Ready

## Related Issues
fixes:  https://github.com/demisto/etc/issues/23939
related pr: https://github.com/demisto/content/pull/7087
## Description
Added a new function auto_detect_indicator_type for the issue: demisto/etc#23939

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -252,6 +253,65 @@ def ip_to_indicator_type(ip):
...
+      :param indicator_value: The indicator whose type we want to check. (required)
+
+      :return: The type of the indicator.
+      :rtype: ``str``
+    """
+    try:
+        import tldextract
+    except Exception:
+        raise Exception("Missing tldextract module, In order to use the auto detect function please use a docker"
+                        " image with it installed such as: demisto/jmespath:1.0.0.6980")
```
Comment (bakatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:270]: the tag (1.0.0.6980 is redundant,
```suggestion
                        " image with it installed such as the latest tag of: demisto/jmespath")
```

File: Packs/Base/ReleaseNotes/1_0_2.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+- __CommonServerPython__
+Added new function **auto_detect_indicator_type** to indicate the type of indicator.
```
Comment (amshamah419) [inline Packs/Base/ReleaseNotes/1_0_2.md:4]:
```suggestion
  - Added new function **auto_detect_indicator_type** to indicate the type of indicator.
```

**Comment by orhovy** [comment]
Force merge is needed  - as a result of this issue: https://github.com/demisto/etc/issues/24945

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.yml [hunk 3]
```diff
@@ -13,7 +13,6 @@ system: true
 scripttarget: 0
 dependson: {}
 timeout: 0s
-alt_dockerimages: # used for unit testing
-  - demisto/python3:3.7.3.286
+dockerimage: demisto/python3:3.8.3.8494   # used for unit testing
```
Comment (bakatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.yml:16]: is the `alt_dockerimages` still needed? or can we change that? @glicht

Comment (glicht) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.yml:16]: Yes it is needed.

Comment (glicht) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.yml:16]: Also don't add the docker image tag as then we are NOT using the default image.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py [hunk 4]
```diff
@@ -1822,5 +1822,5 @@ def test_auto_detect_indicator_type(indicator_value, indicatory_type):
         try:
             auto_detect_indicator_type(indicator_value)
         except Exception as e:
-            assert str(e) == "Missing tldextract module, In order to use the auto detect function please use a docker" \
-                        " image with it installed such as: demisto/jmespath"
+            assert str(e) == "Missing tldextract module, In order to use the auto detect function please" \
```
Comment (bakatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:1825]: you can use in and a partial string match if you wish.

**Pull request merged**